src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/MicrosoftExtensionsLogging/MicrosoftLoggingWrapper.cs

```diff
@@ -71,27 +75,24 @@ private void RecordLogMessage(MethodCall methodCall, MEL.ILogger logger, IAgent
 
             try
             {
-                // We are trying to access this property:
-                // logger.Loggers[0].Logger.ScopeProvider
+                // MEL keeps an array of scope handlers. We can ask one of them for the current scope data.
 
-                // Get the array of Loggers (logger.Loggers[])
-                var getLoggersArrayFunc = _getLoggersArray ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<dynamic>(logger.GetType(), "Loggers");
+                // Get the array of ScopeLoggers (logger.ScopeLoggers[])
+                var getLoggersArrayFunc = _getLoggersArray ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<dynamic>(logger.GetType(), "ScopeLoggers");
                 var loggers = getLoggersArrayFunc(logger);
 
-                // Get the first logger in the array (logger.Loggers[0])
+                // Get the first ScopeLogger in the array (logger.ScopeLoggers[0])
+                // If there is more than one scope logger, they've all received the same data, so the first
+                // one should be fine
                 object firstLogger = loggers.GetValue(0);
```

Comment:
Is it possible that this array has a length of 0?

Reply:
It's a bit confusing, but as far as I can tell, no. The ILogger still seems to be stored as a scope logger, even if it isn't one (There's an integration test that does this). Worst case, this will throw and be caught, resulting in us disabling trying to read scope data going forward. If there are no scope loggers, that would be correct since we wouldn't have any way to get the data.